### PR TITLE
cephadm/box: default add hosts

### PR DESCRIPTION
**Pull request by pereman2**
Now hosts are added to the cluster by default and osds will be created rotating each host. Creating osds takes a while because mons will take some time to be added in new hosts and because of some delay in services.

What you'll see now is the usual behaviour but at the end, there will be a loop executing `osd add` until all of them are added.

Default changes from 1 osd to 3 and from 1 host to 3.

UPDATE: by default no hosts are deployed and osds, with the `box cluster start --expanded` option you can enable this but it will take more time to deploy the cluster. In my machine it takes around 7 mins avg for example (still good tho)
Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by nizamial09 — COMMENTED** [review]
@pereman2 I am seeing 4 host coming up in the cluster and this command goes on an on without ever successfully finishes
```
Running command: CEPHADM_IMAGE=quay.ceph.io/ceph-ci/ceph:master cephadm --verbose shell --config /etc/ceph/ceph.conf --keyring /etc/ceph/ceph.keyring -- ceph orch daemon add osd "574cb221f7da:vg1/lv0"
Running command: CEPHADM_IMAGE=quay.ceph.io/ceph-ci/ceph:master cephadm --verbose shell --config /etc/ceph/ceph.conf --keyring /etc/ceph/ceph.keyring -- ceph orch daemon add osd "574cb221f7da:vg1/lv0"
```
Screenshot from 2022-02-15 12-25-58


Also, instead of making 3 hosts and 3 osds a default, I would prefer something like we have in the kcli script Alfonso did. A `--dev` and `--expanded` flag. `--dev` will just deploy the cluster with a single host and `--expanded` will deploy with 3 hosts and osds. What do you think?

**Comment by pereman2** [comment]
@nizamial09 yeah osds takes too much time (and failed attempts) because of long time waiting for services to deploy. I'll do the `--dev` and `--expanded` option.

File: doc/dev/cephadm/developing-cephadm.rst [hunk 1]
```diff
@@ -264,9 +264,15 @@ In order to setup Cephadm's box run::
 
 .. note:: It is recommended to run box with verbose (-v).
 
-After getting all needed images we can run::
+After getting all needed images we can create a simple cluster with::
```
Comment (epuertat) [inline doc/dev/cephadm/developing-cephadm.rst:267]: I'd emphasize the fact that this cluster doesn't have any hosts or OSDs.

File: doc/dev/cephadm/developing-cephadm.rst [hunk 2]
```diff
@@ -264,9 +264,15 @@ In order to setup Cephadm's box run::
...
 
-  sudo box -v cluster start --osds 3 --hosts 3
+  sudo box -v cluster start
+
+If you want to deploy the cluster with more osds and hosts::
+  # 3 osds and 3 hosts by default
+  sudo box -v cluster start --extended
+  # explicitly change number of hosts and osds
+  sudo box -v cluster start --extended --osds 5 --hosts 5
```
Comment (epuertat) [inline doc/dev/cephadm/developing-cephadm.rst:275]: What happens if I write `--osds 5 --hosts 5` without the `--extended` flag (BTW, shouldn't it be `--expanded`)? I'm just trying to understand the semantics of the flags.

**Review by epuertat — APPROVED** [review]

**Comment by epuertat** [comment]
BTW I tested and it worked ok! 🥳  I couldn't measure how long it takes to complete the start stage... And then my desktop "mysteriously" crashed 🤕

File: doc/dev/cephadm/developing-cephadm.rst (hunk 2 above)
Comment (pereman2) [reply doc/dev/cephadm/developing-cephadm.rst:275]: It won't make a change, I'll add a comment on that.

**Pull request merged**